src/Pirouette/Term/Syntax/Base.hs

```diff
@@ -290,6 +299,13 @@ data Definition lang
   | DTypeDef (TypeDef lang)
   deriving (Eq, Ord, Show, Data)
 
+instance (LanguageBuiltins lang) => AlphaEq (Definition lang) where
+  DFunDef f1 ~==~ DFunDef f2 = f1 == f2 -- TODO shall we do this modulo α-equivalence too?
```

Comment:
Function definitions and types `Eq` instance is alpha-equivalence because we're using De Bruijn indices and `Ann`'s are ignored.

Reply:
Sorry, that's a vague comment — I was thinking about doing the same for the `Ann` type that's used in terms (and thus in function definitions), and that right now is defined to relate any two elements of a particular type.

Probably not worth it though. LMK if you agree, and I'll remove the `TODO` and merge.

Reply:
I wouldn't change the definition of equality for types and terms; the whole purpose of `Ann` is precisely to be just an annotation and not interfere with anything. For types and terms, `Eq` and `Ord` are compatible because they're both derived. 

I'm happy with removing the comment and merging :)

Reply:
Maybe something like the suggestion below would be nice, so we don't forget